File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    hl = 0
    val = 0
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        self.find(root,1)
        return self.val
        
    def find(self,root,lvl):
        if not root : return 
        if self.hl < lvl :
            self.val = root.val
            self.hl = lvl 
        self.find(root.left,lvl+1)
        self.find(root.right,lvl+1)

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        queue<TreeNode *> q;
        int ans ;
        vector<int> temp;
        q.push(root);
        q.push(nullptr);
        while(!q.empty()){
            TreeNode *node = q.front();
            q.pop();
            if(node==nullptr){
                ans = temp[0];
                temp.clear();
                if(!q.empty()){ q.push(nullptr);}
            }
            else{
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
                temp.push_back(node->val);
            }
        }
        return ans;
    }
};